signingscript/task.py

```diff
@@ -137,68 +143,100 @@ def get_suitable_signing_servers(signing_servers, cert_type, signing_formats):
 
 
 # sign_file {{{1
-async def sign_file(context, from_, cert_type, signing_formats, cert, to=None):
+async def sign_file(context, orig_file, cert_type, signing_formats, cert):
     """Send a file to the signing server to sign, then retrieve the signed file.
 
     In post-signing steps, zipalign apks if applicable.
 
     Args:
         context (SigningContext): the signing context
-        from_ (str): the source file to sign
+        orig_file (str): the source file to sign
         cert_type (str): the cert type used to find an appropriate signing server
         signing_formats (str): the formats to sign with
         cert (str): the path to the ssl cert, if applicable
-        to (str, optional): the path to write the signed file to.  If None,
-            overwrite `from_`.  Defaults to None.
 
     Raises:
         FailedSubprocess: on subprocess error while signing.
 
     """
-    from_ = await _execute_pre_signing_steps(context, from_)
-    to = to or from_
     work_dir = context.config['work_dir']
     token = os.path.join(work_dir, "token")
     nonce = os.path.join(work_dir, "nonce")
     signtool = context.config['signtool']
     if not isinstance(signtool, (list, tuple)):
         signtool = [signtool]
-    signing_command = signtool + ["-v", "-n", nonce, "-t", token, "-c", cert]
-    for s in get_suitable_signing_servers(context.signing_servers, cert_type, signing_formats):
-        signing_command.extend(["-H", s.server])
-    for f in signing_formats:
-        signing_command.extend(["-f", f])
-    signing_command.extend(["-o", to, from_])
-    await utils._execute_subprocess(signing_command)
+    files, callback = await _execute_pre_signing_steps(context, orig_file, signing_formats)
+    for from_ in files:
+        log.info("Signing {}...".format(from_))
+        to = from_
+        if callback is not None and not callback(from_):
+            continue
+        signing_command = signtool + ["-v", "-n", nonce, "-t", token, "-c", cert]
+        for s in get_suitable_signing_servers(context.signing_servers, cert_type, signing_formats):
+            signing_command.extend(["-H", s.server])
+        for f in signing_formats:
+            signing_command.extend(["-f", f])
+        signing_command.extend(["-o", to, from_])
```

Comment:
Sure.  It's a list, so we have to do `signing_command = base_command[:]`, which I never really liked syntax-wise, but it makes sense.